Importing logic app gives Error: `inputs` was nil for HTTP Trigger Trigger

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.11.4

### AzureRM Provider Version

4.47.0

### Affected Resource(s)/Data Source(s)

azurerm_logic_app_trigger_http_request

### Terraform Configuration Files

```hcl
# also happens when using "terraform import" command instead of plan with import blocks
import {
  id = "/subscriptions/111-111-111/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/reboot-when-unhealthy/triggers/When_an_HTTP_request_is_received"
  to = azurerm_logic_app_trigger_http_request.main
}
resource "azurerm_logic_app_trigger_http_request" "main" {
  name = "When_an_HTTP_request_is_received"
  logic_app_id = azurerm_logic_app_workflow.main.id

  schema = jsonencode({
  })
}
```

### Debug Output/Panic Output

```shell
azurerm_logic_app_trigger_http_request.main: Refreshing state... [id=/subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow/triggers/When_an_HTTP_request_is_received]
2025-11-10T14:19:19.596-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Preparing arguments for /subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow: Trigger "When_an_HTTP_request_is_received"
2025-11-10T14:19:19.596-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Locking "azurerm_logic_app.my-workflow"
2025-11-10T14:19:19.596-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Locked "azurerm_logic_app.my-workflow"
2025-11-10T14:19:19.596-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] AzureRM Request:
2025-11-10T14:19:19.596-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: GET /subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow?api-version=2019-05-01 HTTP/1.1
2025-11-10T14:19:19.597-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Host: management.azure.com
2025-11-10T14:19:19.602-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 workflows/2019-05-01) HashiCorp Terraform/1.11.4 (+https://www.terraform.io) terraform-provider-azurerm/dev pid-222c6c49-1b0a-5959-a213-6608f9eb8820
2025-11-10T14:19:19.602-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Content-Type: application/json; charset=utf-8
2025-11-10T14:19:19.603-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Correlation-Request-Id: 40322bc9-6aa9-34e5-c359-bce2ffbdb961
2025-11-10T14:19:19.603-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Accept-Encoding: gzip
2025-11-10T14:19:19.603-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe
2025-11-10T14:19:19.603-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe
2025-11-10T14:19:19.603-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] GET https://management.azure.com/subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow?api-version=2019-05-01
2025-11-10T14:19:19.979-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow?api-version=2019-05-01:
2025-11-10T14:19:19.979-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: HTTP/2.0 200 OK
2025-11-10T14:19:19.979-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Content-Length: 2839
2025-11-10T14:19:19.979-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Cache-Control: no-cache
2025-11-10T14:19:19.979-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Content-Type: application/json; charset=utf-8
2025-11-10T14:19:19.979-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Date: Mon, 10 Nov 2025 19:19:19 GMT
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Expires: -1
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Pragma: no-cache
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Strict-Transport-Security: max-age=31536000; includeSubDomains
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Vary: Accept-Encoding
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Cache: CONFIG_NOCACHE
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Content-Type-Options: nosniff
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Correlation-Request-Id: 40322bc9-6aa9-34e5-c359-bce2ffbdb961
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Ratelimit-Remaining-Subscription-Resource-Requests: 1099
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Request-Id: canadacentral:8bf33b5d-ee91-40a6-9405-289d0dbc58bf
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Routing-Request-Id: CANADACENTRAL:20251110T191919Z:eefebd46-54b6-42a9-8243-eacb6efe00b4
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Msedge-Ref: Ref A: E70103F363B9437A915416F8CC88675A Ref B: YTO221090812027 Ref C: 2025-11-10T19:19:19Z
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe
2025-11-10T14:19:19.980-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: {"properties":{"provisioningState":"Succeeded","createdTime":"2025-10-07T14:28:32.0886077Z","changedTime":"2025-11-10T18:50:44.4106686Z","state":"Enabled","version":"08584388062410663521","accessEndpoint":"redacted","definition":{"$schema":"https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#","contentVersion":"1.0.0.0","parameters":{"$connections":{"defaultValue":{},"type":"Object"}},"triggers":{"When_an_HTTP_request_is_received":{"type":"Request","kind":"Http"}},"actions":{"Queue_a_new_build":{"runAfter":{},"type":"ApiConnection","inputs":{"body":{"sourceBranch":"uat"},"host":{"connection":{"name":"@parameters('$connections')['visualstudioteamservices']['connectionId']"}},"method":"post","path":"/@{encodeURIComponent('CKAN_DC')}/_apis/build/builds","queries":{"account":"aafc","buildDefId":"305"}}}}},"parameters":{"$connections":{"value":{"visualstudioteamservices":{"connectionId":"/subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Web/connections/visualstudioteamservices","connectionName":"visualstudioteamservices","id":"/subscriptions/redacted-subscription-id/providers/Microsoft.Web/locations/canadacentral/managedApis/visualstudioteamservices"}}}},"endpointsConfiguration":{"workflow":{"outgoingIpAddresses":[{"address":"redacted"},{"address":"redacted"},{"address":"redacted"},{"address":"redacted"},{"address":"redactedaddress":"redacted"},{"address":"redacted"},{"address":"redacted"},{"address":"redactedaddress":"redacted"}],"accessEndpointIpAddresses":[{"address":"redacted"},{"address":"redacted"},{"address":"redactedaddress":"redacted"},{"address":"redacted"},{"address":"redacted"},{"address":"redactedaddress":"redacted"}]},"connector":{"outgoingIpAddresses":[{"address":"redacted"},{"address":"redacted"},{"address":"redacted/28"},{"address":"redacted/27"},{"address":"redacted/27"},{"address":"redacted/28"},{"address":"redacted"},{"address":"redacted"}]}}},"id":"/subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow","name":"my-workflow","type":"Microsoft.Logic/workflows","location":"canadacentral","tags":{}}
2025-11-10T14:19:19.981-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Unlocking "azurerm_logic_app.my-workflow"
2025-11-10T14:19:19.981-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Unlocked "azurerm_logic_app.my-workflow"
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] trigger type is Request
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Preparing arguments for Logic App Workspace "my-workflow" (Resource Group "my-rg") trigger "When_an_HTTP_request_is_received"
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Locking "azurerm_logic_app.my-workflow"
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Locked "azurerm_logic_app.my-workflow"
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] AzureRM Request:
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: POST /subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow/triggers/When_an_HTTP_request_is_received/listCallbackUrl?api-version=2019-05-01 HTTP/1.1
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Host: management.azure.com
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 workflowtriggers/2019-05-01) HashiCorp Terraform/1.11.4 (+https://www.terraform.io) terraform-provider-azurerm/dev pid-222c6c49-1b0a-5959-a213-6608f9eb8820
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Content-Length: 0
2025-11-10T14:19:19.982-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Content-Type: application/json; charset=utf-8
2025-11-10T14:19:19.983-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Correlation-Request-Id: 40322bc9-6aa9-34e5-c359-bce2ffbdb961
2025-11-10T14:19:19.983-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Accept-Encoding: gzip
2025-11-10T14:19:19.983-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe
2025-11-10T14:19:19.983-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe
2025-11-10T14:19:19.983-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] POST https://management.azure.com/subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow/triggers/When_an_HTTP_request_is_received/listCallbackUrl?api-version=2019-05-01
2025-11-10T14:19:20.258-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow/triggers/When_an_HTTP_request_is_received/listCallbackUrl?api-version=2019-05-01:
2025-11-10T14:19:20.259-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: HTTP/2.0 200 OK
2025-11-10T14:19:20.259-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Content-Length: 622
2025-11-10T14:19:20.259-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Cache-Control: no-cache
2025-11-10T14:19:20.259-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Content-Type: application/json; charset=utf-8
2025-11-10T14:19:20.258-0500 [ERROR] vertex "import azurerm_logic_app_trigger_http_request.main result" error: `inputs` was nil for HTTP Trigger Trigger (Subscription: "redacted-subscription-id"
Resource Group Name: "my-rg"
Workflow Name: "my-workflow"
Trigger Name: "When_an_HTTP_request_is_received")
2025-11-10T14:19:20.260-0500 [ERROR] vertex "azurerm_logic_app_trigger_http_request.main (import id \"/subscriptions/redacted-subscription-id/resourceGroups/my-rg/providers/Microsoft.Logic/workflows/my-workflow/triggers/When_an_HTTP_request_is_received\")" error: `inputs` was nil for HTTP Trigger Trigger (Subscription: "redacted-subscription-id"
Resource Group Name: "my-rg"
Workflow Name: "my-workflow"
Trigger Name: "When_an_HTTP_request_is_received")
2025-11-10T14:19:20.260-0500 [ERROR] vertex "azurerm_logic_app_trigger_http_request.main (expand)" error: `inputs` was nil for HTTP Trigger Trigger (Subscription: "redacted-subscription-id"
Resource Group Name: "my-rg"
Workflow Name: "my-workflow"
Trigger Name: "When_an_HTTP_request_is_received")
2025-11-10T14:19:20.259-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Date: Mon, 10 Nov 2025 19:19:19 GMT
2025-11-10T14:19:20.261-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Expires: -1
2025-11-10T14:19:20.261-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Pragma: no-cache
2025-11-10T14:19:20.262-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Strict-Transport-Security: max-age=31536000; includeSubDomains
2025-11-10T14:19:20.262-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Vary: Accept-Encoding
2025-11-10T14:19:20.262-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Cache: CONFIG_NOCACHE
2025-11-10T14:19:20.262-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Content-Type-Options: nosniff
2025-11-10T14:19:20.262-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Correlation-Request-Id: 40322bc9-6aa9-34e5-c359-bce2ffbdb961
2025-11-10T14:19:20.262-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Operation-Identifier: tenantId=9da98bb1-1857-4cc3-8751-9a49e35d24cd,objectId=2a66ec41-3646-4e92-8935-dadf5999b650/canadacentral/9ac1adc3-1347-4b26-9452-cca7b88cce12
2025-11-10T14:19:20.267-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Ratelimit-Remaining-Subscription-Global-Writes: 2999
2025-11-10T14:19:20.267-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Ratelimit-Remaining-Subscription-Writes: 199
2025-11-10T14:19:20.268-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Request-Id: canadacentral:236bea88-0b16-42ab-8650-169bdc347abf
2025-11-10T14:19:20.268-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Ms-Routing-Request-Id: CANADACENTRAL:20251110T191920Z:eabf3617-3d31-4c46-b637-a9e71d2d92ea
2025-11-10T14:19:20.269-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: X-Msedge-Ref: Ref A: 5FCD1A6DBE3948DF8507E128A910E175 Ref B: YTO221100611025 Ref C: 2025-11-10T19:19:20Z
2025-11-10T14:19:20.269-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe
╷
│ Error: `inputs` was nil for HTTP Trigger Trigger (Subscription: "redacted-subscription-id"
│ Resource Group Name: "my-rg"
│ Workflow Name: "my-workflow"
│ Trigger Name: "When_an_HTTP_request_is_received")
│
│
╵

2025-11-10T14:19:20.269-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: {"value":"https://prod-13.canadacentral.logic.azure.com:443/workflows/jkljkljkl/triggers/When_an_HTTP_request_is_received/paths/invoke?api-version=2019-05-01&sp=%2Ftriggers%2FWhen_an_HTTP_request_is_received%2Frun&sv=1.0&sig=abc","method":"POST","basePath":"https://prod-13.canadacentral.logic.azure.com/workflows/jkljkljkl/triggers/When_an_HTTP_request_is_received/paths/invoke","queries":{"api-version":"2019-05-01","sp":"/triggers/When_an_HTTP_request_is_received/run","sv":"1.0","sig":"abc123"}}
2025-11-10T14:19:20.271-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Unlocking "azurerm_logic_app.my-workflow"
2025-11-10T14:19:20.272-0500 [DEBUG] provider.terraform-provider-azurerm_v4.47.0_x5.exe: [DEBUG] Unlocked "azurerm_logic_app.my-workflow"
2025-11-10T14:19:20.272-0500 [ERROR] provider.terraform-provider-azurerm_v4.47.0_x5.exe: Response contains error diagnostic:
  diagnostic_summary=
  | `inputs` was nil for HTTP Trigger Trigger (Subscription: "redacted-subscription-id"
  | Resource Group Name: "my-rg"
  | Workflow Name: "my-workflow"
  | Trigger Name: "When_an_HTTP_request_is_received")
   @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_severity=ERROR tf_proto_version=5.10 tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=0192c0d2-714a-4709-b7d8-3e27c9c2fbd7 tf_resource_type=azurerm_logic_app_trigger_http_request diagnostic_detail="" tf_rpc=ReadResource timestamp=2025-11-10T14:19:20.257-0500
2025-11-10T14:19:20.273-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-11-10T14:19:20.292-0500 [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/hashicorp/azurerm/4.47.0/windows_amd64/terraform-provider-azurerm_v4.47.0_x5.exe id=45784
2025-11-10T14:19:20.293-0500 [DEBUG] provider: plugin exited
```

### Expected Behaviour

Import should succeed for http triggers which take no inputs.

### Actual Behaviour

Import failed when reading remote resource.

### Steps to Reproduce

1. Create a logic app
2. Create an http trigger that takes no arguments
3. Import the trigger

### Important Factoids

_No response_

### References

https://github.com/hashicorp/terraform-provider-azurerm/blob/550bd7da3f98d851f3894a2162cb263eb85d140a/internal/services/logic/logic_app_trigger_http_request_resource.go#L163

--- a/internal/services/logic/logic_app_action_http_resource.go
+++ b/internal/services/logic/logic_app_action_http_resource.go
@@ -213,7 +213,6 @@ func resourceLogicAppActionHTTPRead(d *pluginsdk.ResourceData, meta interface{})
 
 	v := action["inputs"]
 	if v != nil {
-
 		inputs, ok := v.(map[string]interface{})
 		if !ok {
 			return fmt.Errorf("parsing `inputs` for HTTP Action %s", id)

